examples/local/vstream_client.go

```diff
@@ -82,17 +72,372 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	err = readEvents(ctx, reader, time.Duration(flags.HeartbeatInterval)*time.Second)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// getLastVgtid retrieves the last vgtid processed by the client, so that it can resume from that position.
+func getLastVgtid(ctx context.Context) (*binlogdatapb.VGtid, error) {
+	var vgtid binlogdatapb.VGtid
+
+	// if storeLastVgtid was implemented, you would retrieve the last vgtid and unmarshal it here
+	// err := json.Unmarshal([]byte{}, &vgtid)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	streamCustomer := true
+	if streamCustomer {
+		vgtid = binlogdatapb.VGtid{
+			ShardGtids: []*binlogdatapb.ShardGtid{{
+				Keyspace: "customer",
+				Shard:    "-80",
+				// Gtid "" is to stream from the start, "current" is to stream from the current gtid
+				// you can also specify a gtid to start with.
+				Gtid: "", // "current"  // "MySQL56/36a89abd-978f-11eb-b312-04ed332e05c2:1-265"
+			}, {
+				Keyspace: "customer",
+				Shard:    "80-",
+				Gtid:     "",
+			}},
+		}
+	} else {
+		vgtid = binlogdatapb.VGtid{
+			ShardGtids: []*binlogdatapb.ShardGtid{{
+				Keyspace: "commerce",
+				Shard:    "0",
+				Gtid:     "",
+			}},
+		}
+	}
+
+	return &vgtid, nil
+}
+
+// storeLastVgtid stores the last vgtid processed by the client, so that it can resume from that position on restart.
+// Storing a json blob in a database is just one way to do this, you could put it anywhere.
+func storeLastVgtid(ctx context.Context, vgtid *binlogdatapb.VGtid) error {
+	_, err := json.Marshal(vgtid)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Is storing a full json blob the right thing to do? Do we really only want a subset if we want vstream to handle reshards?